src/arrayMethodSort.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare =
+      compareFunction ||
+      function (x, y) {
+        if (String(x) > String(y)) {
+          return 1;
+        }
+
+        if (String(x) < String(y)) {
+          return -1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
+
+    return this;
```

Comment:
While implementing a sort method using the bubble sort algorithm is acceptable, it is known for its inefficiency on large datasets. Consider implementing a more efficient sorting algorithm like quicksort or mergesort for better performance on large arrays.